### PR TITLE
Install deps in each package's isolated location

### LINKED ISSUE #12: Install dependencies to the package's install location
Instead of installing Python dependencies globally, it would be neat to put it under Colcon's `install/` directory for that package. This would provide the kind of dependency isolation that users expect when using Poetry. The main sticking point here is that it would require creating and laying out the `install/` directory before Colcon does, since dependency installation should come before building. That might cause problems.

The dependency installation command should support Colcon's `--install-base` and `--merge-install` arguments.

### PR DESCRIPTION
**Pull request by velovix**
Fixes #12 

## To-do before merge

- [x] Do a full scale test with this on actual hardware to ensure everything is installed

### CODE REVIEW

File: colcon_poetry_ros/project.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+from typing import List, Set
+
+import toml
+
+
+class NotAROSPoetryProjectError(Exception):
+    """The given directory does not point to a ROS Poetry project"""
+
+
+class ROSPoetryProject:
```
Comment (velovix) [inline colcon_poetry_ros/project.py:13]: Pretty much all of this code has been migrated from elsewhere and refactored into a class.

**Review by apockill — APPROVED** [review]
Yo this is exciting! Have you tested it on the project? :eyes:

File: colcon_poetry_ros/dependencies/install.py [hunk 2]
```diff
@@ -1,73 +1,123 @@
...
+                install_base /= project.name
+
+            subprocess.run(
+                [
+                    "python3",
+                    "-m",
+                    "pip",
+                    "install",
+                    # I don't understand why, but providing this fixes:
+                    # https://github.com/pypa/pip/issues/9644
```
Comment (apockill) [inline colcon_poetry_ros/dependencies/install.py:38]: lol

**Comment by velovix** [comment]
@apockill I have! But not super extensively. It passes a smoke test of starting the system up, but that's as much as I can say for now. We should definitely run this for real tomorrow before merging.

**Review by apockill — APPROVED** [review]

File: colcon_poetry_ros/dependencies/install.py (hunk 2 above)
Comment (velovix) [reply colcon_poetry_ros/dependencies/install.py:38]: This crap is very weird.

**Pull request merged**